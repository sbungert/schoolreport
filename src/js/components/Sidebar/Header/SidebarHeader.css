/* Sidebar Header */
/* -------------- */

.SidebarHeader {
  position: absolute;
	display: flex;
	flex-direction: row;
	justify-content: flex-start;
	align-items: center;
  top: 0px;
  left: 0px;
  margin: 0px;
  padding: 0px 20px;
  width: 100%;
  height: var(--sidebar-header-height);
  background: var(--col-brand-dark);
  border-bottom: 1px solid var(--col-darken-1);
}

.SidebarHeader::after {
  opacity: 0.1;
  content: " ";
  position: absolute;
  bottom: -11px;
  left: 0px;
  width: 100%;
  height: 10px;
  display: block;

      /* http://colorzilla.com/gradient-editor/#000000+0,000000+100&1+0,0+100 */
  background: -moz-linear-gradient(top, rgba(0,0,0,1) 0%, rgba(0,0,0,0) 100%);
  background: -webkit-linear-gradient(top, rgba(0,0,0,1) 0%,rgba(0,0,0,0) 100%);
  background: linear-gradient(to bottom, rgba(0,0,0,1) 0%,rgba(0,0,0,0) 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#000000', endColorstr='#00000000',GradientType=0 );
}

.SidebarHeader__headline {
  margin: 0px;
  font-size: var(--font-size);
  font-weight: normal;
	overflow: hidden;
  display: block;
  width: 100%;
	text-overflow: ellipsis;
	white-space: nowrap;
  color: var(--col-white);
}

@media screen and (min-width: 1024px) {
    .SidebarHeader {
      border-right: 1px solid var(--col-sidebar-border);
    }
}
