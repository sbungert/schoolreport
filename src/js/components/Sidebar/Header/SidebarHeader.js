// @flow

import React, { PureComponent } from 'react';
import './SidebarHeader.css';

type Props = {
  title: string,
  search: boolean,
  sort: boolean,
};


/**
* Sidebar header.
*/
class SidebarHeader extends PureComponent<Props> {
  static defaultProps = {
      title: '',
      search: true,
      sort: true,
   };

  props: Props;

  getActions() {
    let actions = [];

    //if (this.props.search) actions.push(<span className="SidebarHeader__action" data-action="search">S1</span>);
    //if (this.props.sort) actions.push(<span className="SidebarHeader__action" data-action="sort">S2</span>);

    return actions;
  }

  render() {
    return (
      <header className="SidebarHeader">
        <h1 className="SidebarHeader__headline">{this.props.title}</h1>
      </header>
    )
  }
}

export default SidebarHeader;
