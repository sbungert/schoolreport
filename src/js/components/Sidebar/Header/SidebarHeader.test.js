// @flow

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SidebarHeader from './SidebarHeader';

describe('<SidebarHeader />', () => {
  const props = { title: 'Test title' };

  test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SidebarHeader {...props} />, div);
  });

  test('Props render correctly', () => {
    const testRender = renderer.create(<SidebarHeader {...props} />).toJSON();
    expect(testRender).toMatchSnapshot();
  });
});
