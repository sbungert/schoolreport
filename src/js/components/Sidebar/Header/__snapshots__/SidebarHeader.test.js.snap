// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarHeader /> Props render correctly 1`] = `
<header
  className="SidebarHeader"
>
  <h1
    className="SidebarHeader__headline"
  >
    Test title
  </h1>
</header>
`;
